primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -652,6 +651,344 @@ where
     }
 }
 
+#[cfg(feature = "icicle")]
+pub(crate) mod icicle {
+    use super::*;
+    use crate::icicle_deps::{curves::*, *};
+    use ark_ec::{short_weierstrass::Affine, CurveConfig};
+    use ark_ff::BigInteger;
+
+    /// Trait for GPU-accelerated PCS.commit APIs
+    pub trait GPUCommit<E, C>
+    where
+        C: IcicleCurve + MSM<C>,
+        C::ScalarField: ArkConvertible<ArkEquivalent = E::ScalarField>,
+        C::BaseField: ArkConvertible<ArkEquivalent = <C::ArkSWConfig as CurveConfig>::BaseField>,
+        <C::ArkSWConfig as CurveConfig>::BaseField: PrimeField,
+        E: Pairing<G1Affine = Affine<<C as IcicleCurve>::ArkSWConfig>>,
+    {
+        /// The full cycle of computing poly-commit on GPU
```

Comment:
Should we also add another API that includes all steps except for loading prover params? This gives a convenient API if the user wants to commit multiple polynomials but doesn't want to design the computing flow herself.

Another option is to remove the loading_prover_params step in this API and warn the user to call `load_prover_params` before calling this API.

Either option above works for me. (I slightly prefer adding another API)

Reply:
definitely, adding another api is better, we want users not to figure out how to use icicle themselves.

@mrain also suggested the same additional handy API offline. I'll add it.